File: MyGame/app/src/main/java/com/example/mygame/domain/IVisitor.kt
package com.example.mygame.domain

import com.example.mygame.domain.bonus.Jetpack
import com.example.mygame.domain.bonus.Shield
import com.example.mygame.domain.bonus.Spring
import com.example.mygame.domain.bullet.Bullet
import com.example.mygame.domain.platform.Platform
import com.example.mygame.domain.player.Player

interface IVisitor {
    fun visit(platform: Platform)
    fun visit(player: Player)
    fun visit(spring: Spring)
    fun visit(shield: Shield)
    fun visit(jetpack: Jetpack)
    fun visit(enemy: Enemy)
    fun visit(bullet: Bullet)
}

File: GameServer/src/Collision/IVisitor.php
<?php

namespace App\Collision;

use App\Model\Enemy;
use App\Model\Bullet;
use App\Model\Player;
use App\Model\Platform;
use App\Model\Bonus\Shield;
use App\Model\Bonus\Spring;
use App\Model\EndZone;

interface IVisitor
{
    function visitPlatform(Platform $platform): void;
    function visitPlayer(Player $player): void;
    function visitSpring(Spring $spring): void;
    function visitEndZone(EndZone $endZone): void;
}
